packages/babel-helper-skip-transparent-expression-wrappers/src/index.ts

```diff
@@ -0,0 +1,49 @@
+import * as t from "@babel/types";
+import { NodePath } from "@babel/traverse";
+
+export type TransparentExprWrapper =
+  | t.TSAsExpression
+  | t.TSTypeAssertion
+  | t.TSNonNullExpression
+  | t.TypeCastExpression
+  | t.ParenthesizedExpression;
+
+// A transparent expression wrapper is an AST node that most plugins will wish
+// to skip, as its presence does not affect the behaviour of the code. This
+// includes expressions used for types, and extra parenthesis. For example, in
+// (a as any)(), this helper can be used to skip the TSAsExpression when
+// determining the callee.
+export function isTransparentExprWrapper(
+  node: t.Node,
+): node is TransparentExprWrapper {
+  return (
+    t.isTSAsExpression(node) ||
+    t.isTSTypeAssertion(node) ||
+    t.isTSNonNullExpression(node) ||
+    t.isTypeCastExpression(node) ||
+    t.isParenthesizedExpression(node)
+  );
+}
+
+type ExprNodeOrPath = t.Expression | NodePath<t.Expression>;
+
+export function skipTransparentExprWrappers(expr: t.Expression): t.Expression;
+
+export function skipTransparentExprWrappers(
+  expr: NodePath<t.Expression>,
+): NodePath<t.Expression>;
```

Comment:
We could also just expose two functions (`skipTransparentExprWrappers` and `skipTransparentExprWrappersNodes`); lets see what other team members think.

Reply:
Agree.

@lightmare Can you split this PR? One for JS->TS and the other with a new API? So we can merge the JS->TS in patch releases.